### PR TITLE
Add Parameterization documentation

### PR DESCRIPTION
**Pull request by rshih32**
Explain how to configure trulens

### CODE REVIEW

File: docs/attribution_parameterization.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Example use case: Say you have a concat layer and you want to explain the 2 concatenated tensors. The concat operation is not usually a layer tracked by the model. If you try the 'in' anchor of the layer after the operation, you get a single tensor with all the information you need.
+
+**What is a Quantity of Interest (QoI)?**
+A QoI is a scalar number that is being explained. 
+
+Eg: With saliency maps, you get `dx/dy` (i.e the effect of input on output). `y` in this case is the QoI scalar. It is usually the output of a neuron, but could be a sum of multiple neurons.
+
+**What is an attribution?**
+An attribution is a numerical number associated with every element in a tensor that explains a QoI.
```
Comment (reetika-roy) [inline docs/attribution_parameterization.md:24]: "numerical number" > "numerical value"?

File: docs/attribution_parameterization.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+**Use case:** Explain the input given the output.  
+**Cuts needed:** Trulens Defaults.  
+**Attribution Cut** (The tensor we would like to assign importance) → InputCut aka model args/kwargs  
+**QoI Cut** (The tensor that we are interested to explain) → OutputCut
+  
+#### Case 2: The QoI Cut
+
+Now suppose you want to explain some internal layer’s output (intermediate output)  
+i.e how the input is affecting the output at some intermediate layer.
```
Comment (reetika-roy) [inline docs/attribution_parameterization.md:50]: Was this new line intended?

File: docs/attribution_parameterization.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+**What is a tensor?**
+A tensor is a multidimensional object that can be model inputs, or layer activations.
+
+**What is a layer?**
+A layer is a set of neurons that can be thought of as a function on input tensors. Layer inputs are tensors. Layer outputs are modified tensors.
+
+**What are anchors?**
+Anchors are ways of specifying which tensors you want. You may want the input tensor of a layer, or the output tensor of a layer. 
+
+Example use case: Say you have a concat layer and you want to explain the 2 concatenated tensors. The concat operation is not usually a layer tracked by the model. If you try the 'in' anchor of the layer after the operation, you get a single tensor with all the information you need.
```
Comment (reetika-roy) [inline docs/attribution_parameterization.md:16]: Use `Eg` like everywhere else?

File: docs/attribution_parameterization.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+**Attribution Cut** → InputCut  
+**QoI Cut** → Your logit layer, anchor:'out'
+ 
+#### Case 3: The Attribution Cut
+Now suppose you want to know the attribution of some internal layer on the final output. 
+
+**Use cases:** 
+
+* As a preprocessing step, you drop a feature, so do not need attributions on that.
+* For torch models, model inputs are not tensors. so you'd want the in anchor of the first layer
```
Comment (reetika-roy) [inline docs/attribution_parameterization.md:64]: 'in'

File: docs/attribution_parameterization.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+* As a preprocessing step, you drop a feature, so do not need attributions on that.
+* For torch models, model inputs are not tensors. so you'd want the in anchor of the first layer  
+
+**Cuts needed:** As you want to know the affect of some other layer rather than the input layer, you need to customize the attribution cut.  
+**Model inputs** → InputCut  
+**Attribution Cut** → Your attribution layer (The layer you want to assign importance/attributions with respect to output), anchor:'in'  
+**QoI Cut** → OutputCut
+ 
+#### Case 4: The DoI Cut / Explanation flexibility
```
Comment (reetika-roy) [inline docs/attribution_parameterization.md:71]: Elaborate what `DoI` stands for since this is the first use

**Review by reetika-roy — COMMENTED** [review]
Minor notes on some of the text

File: docs/attribution_parameterization.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+## Attribution Parameterization 
+
+Attributions for different models and use cases can range from simple to more complex. This will help see how to set the various parameters to get what you need.
+
+### Basic Definitions and Terminology
+
+**What is a tensor?**
+A tensor is a multidimensional object that can be model inputs, or layer activations.
```
Comment (piotrm0) [inline docs/attribution_parameterization.md:8]: Need to expand on this to make anchors less confusing. Tensor may mean an operation/function that has inputs and has outputs.

File: docs/attribution_parameterization.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+Anchors are ways of specifying which tensors you want. You may want the input tensor of a layer, or the output tensor of a layer. 
+
+Eg: Say you have a concat layer and you want to explain the 2 concatenated tensors. The concat operation is not usually a layer tracked by the model. If you try the 'in' anchor of the layer after the operation, you get a single tensor with all the information you need.
+
+**What is a Quantity of Interest (QoI)?**
+A QoI is a scalar number that is being explained. 
+
+Eg: With saliency maps, you get `dx/dy` (i.e the effect of input on output). `y` in this case is the QoI scalar. It is usually the output of a neuron, but could be a sum of multiple neurons.
+
+**What is an attribution?**
```
Comment (coreyhu) [inline docs/attribution_parameterization.md:23]: Is there somewhere we can explain the relationship between the DoI and attribution cut?

File: docs/attribution_parameterization.md (hunk 6 above)
Comment (rshih32) [reply docs/attribution_parameterization.md:8]: A tensor is not an operation/function with inputs/outputs, that is a layer. Tensors do not have inputs or outputs. Unless I'm missing something.

Comment (piotrm0) [reply docs/attribution_parameterization.md:8]: Ahh, ok. I re-read it. Seems ok.

**Review by coreyhu — APPROVED** [review]

**Pull request merged**